### PR TITLE
feat: add timezone field at date-time-picker

### LINKED ISSUE #2121: Edit timezone for date fields
## Abstract

When editing dates e.g. in the ledger actions, the timezone is set implicitly from the device you are using.

I would like to be able to customise the timezone directly without having to change the timezone of the device I'm using.

## Motivation

This is useful if you live in a different timezone than when the event took place.

## Specification

I'm not sure if you would use the timezone code e.g. GMT or just an offset for this input field. For the developer to decide.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #2121

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [ ] Add timezone field on date-time-picker
image

### CODE REVIEW

File: frontend/app/src/utils/date.ts [hunk 1]
```diff
@@ -60,6 +60,25 @@ export function convertFromTimestamp(
   return time.format(format);
 }
 
+export function convertDateByTimezone(
+  date: string,
+  dateFormat: DateFormat = DateFormat.DateMonthYearHourMinuteSecond,
+  fromTimestamp: string = dayjs.tz.guess(),
+  toTimestamp: string = dayjs.tz.guess()
```
Comment (kelsos) [inline frontend/app/src/utils/date.ts:67]: Wouldn't this be `fromTimezone` and `toTimezone?

**Comment by kelsos** [comment]
At least in my case, something seems to be off with the detection
image

The guessed timezone looks fine, and also the offset seems to be guessed correctly.
image

Any idea why that might be?

**Comment by lukicenturi** [comment]
> At least in my case, something seems to be off with the detection image
> 
> The guessed timezone looks fine, and also the offset seems to be guessed correctly. image
> 
> Any idea why that might be?

There are something wrong with the json data 
image
The offset should be 1 there

**Comment by kelsos** [comment]
> > At least in my case, something seems to be off with the detection image
> > The guessed timezone looks fine, and also the offset seems to be guessed correctly. image
> > Any idea why that might be?
> 
> There are something wrong with the json data image The offset should be 1 there I guess I have to manually check the data

Not exactly, during summer the offset in Central Europe is 2 indeed because of[Daylight Saving Time](https://en.wikipedia.org/wiki/Daylight_saving_time).

While during winter the offset is indeed +1 [Central European Time](https://en.wikipedia.org/wiki/Central_European_Time),
during summer we are using [Central European Summer Time](https://en.wikipedia.org/wiki/Central_European_Summer_Time) which is indeed +2 

Maybe instead of relying on offsets, it would make sense to use the `utc` subentries to narrow down the timezone?

We still have to account for DST wherever it is observed.

**Comment by lukicenturi** [comment]
> > > At least in my case, something seems to be off with the detection image
> > > The guessed timezone looks fine, and also the offset seems to be guessed correctly. image
> > > Any idea why that might be?
> > 
> > 
> > There are something wrong with the json data image The offset should be 1 there I guess I have to manually check the data
> 
> Not exactly, during summer the offset in Central Europe is 2 indeed because of [Daylight Saving Time](https://en.wikipedia.org/wiki/Daylight_saving_time).
> 
> While during winter the offset is indeed +1 [Central European Time](https://en.wikipedia.org/wiki/Central_European_Time), during summer we are using [Central European Summer Time](https://en.wikipedia.org/wiki/Central_European_Summer_Time) which is indeed +2
> 
> Maybe instead of relying on offsets, it would make sense to use the `utc` subentries to narrow down the timezone?
> 
> We still have to account for DST wherever it is observed.

This one updated now, the select options should use `utc` subentries (Europe/Berlin, Asia/Jakarta, etc..)

**Comment by kelsos** [comment]
When deleting the value completely vue logs this error:

image

**Comment by kelsos** [comment]
@lukicenturi there seems to be a forgotten console log, the check failed

**Review by kelsos — APPROVED** [review]
all looks good, thank you

**Pull request merged**